aiida/tools/importexport/dbexport/zip.py

```diff
@@ -21,18 +21,22 @@
 
 class MyWritingZipFile(object):
 
-    def __init__(self, zip_file, fname):
+    def __init__(self, zip_file, fname, encoding):
         self._zipfile = zip_file
         self._fname = fname
         self._buffer = None
+        self._encoding = encoding
 
     def open(self):
         if self._buffer is not None:
             raise IOError('Cannot open again!')
         self._buffer = six.moves.StringIO()
 
     def write(self, data):
-        self._buffer.write(data)
+        if self._encoding:
+            self._buffer.write(str(data.encode(self._encoding)))
```

Comment:
should the buffer type not change to `io.BytesIO` if the user is going to pass bytes to `write`?